### PR TITLE
fix(rich-text): handle list unwrapping

### PR DESCRIPTION
**Pull request by z0al**

### CODE REVIEW

File: packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx:252]: Is the focus lost in this case?

File: packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx:284]: Same question about focus here.

File: packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx:376]: This one too, what happens to the focus?

File: packages/rich-text/src/plugins/List/transforms/toggleList.ts
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/transforms/toggleList.ts:22]: Is this mostly new logic or borrowed from Plate? If the latter, perhaps leave a note.

**Review by DanweDE — COMMENTED** [review]
Impressive :clap: I've still got to do some testing before approval, some comments for now mainly about the cursor in tests.

File: packages/rich-text/src/plugins/List/transforms/toggleList.ts
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/toggleList.ts:22]: The whole file is borrowed from Plate. I made some modifications here and there. I will add some comments.

**Comment by z0al** [comment]
> Impressive clap I've still got to do some testing before approval, some comments for now mainly about the cursor in tests.

@DanweDE Keep in mind that I discovered two issues while working on this that I couldn't fix as part of this PR (will require rewriting the `toggleList` file in this PR, which is mostly borrowed from Plate as of now):

* If you select more than one list item children and press the toolbar toggle it won't do anything
* if you put the cursor in only one list item children and press the toolbar toggle it would unwrap only that child, not the full list item like in the old editor.

I will bring those to the team to discuss the priority.

File: packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx
Comment (z0al) [reply packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx]: fixed

**Review by DanweDE — APPROVED** [review]

**Pull request merged**
